app/controllers/subscribers_controller.rb

```diff
@@ -17,6 +17,24 @@ def create
     end
   end
 
+  def unsubscribe
+    subscriber = Subscriber.find_by(email: params[:unsubscribe][:email])
+
+    if subscriber
+      UnsubscribeMailSender.call(subscriber)
+      redirect_to new_post_path, notice: 'Check your email to find your unsubscribe url!'
+    else
+      redirect_to new_post_path, notice: 'Email not found!'
+    end
+  end
+
+  def cancel_subscription
```

Comment:
I think a better name here would be `confirm unsubscribing` or something like it. Just by the name is a little hard to differ from the one above (`unsubscribe`).